Update contributing md (#59)
inbo/protocolsource: CONTRIBUTING.md

See also [collaborating with renv](https://rstudio.github.io/renv/articles/collaborating.html).
The first time you open the RStudio project `renv` should automatically download and install the appropriate version of `renv` into the project library.
After this has completed, you can use `renv::restore()` to restore the project library locally on your machine.

> In case you need another package than the onces that are installed (see the [DESCRIPTION file](DESCRIPTION)) for this project, proceed as follows:

Reviewer: ```suggestion
In case you need another package than the ones installed (see the [DESCRIPTION file](DESCRIPTION)) for this project, proceed as follows:
```

- `renv::install("nameofthepackage")`
- add the name of the package to the [DESCRIPTION file](DESCRIPTION)
- `renv::snapshot()`: type 'y' in the console when asked to update [the lock file](renv.lock)

> - commit the changed [lock file](renv.lock)

Reviewer: This procedure may cause headache for novices. Perhaps provide a single `protocolhelper` function that accepts a vector of package names and does this all at once?

@hansvancalster @ElsLommelen this should not delay this PR, but if considered a useful idea, be turned into an issue at `protocolhelper`.

Reviewer 2: I have a related issue that I need to address with @hansvancalster: for now the renv packages are installed in the docker images (all three) itself, which has the advantage that the docker containers have a full installation already (which makes the GHA quick), but the disadvantage that we need to build a new docker image after adding new packages (which might be necessary anyway if a package installation needs additional software to run on Ubuntu).  (And also: is this installation of all renv packages necessary for all three GHA, i.e. in which steps do we need to render the protocols?)

In the current situation, we have to take action anyway (rebuild the docker image), so maybe in this case we can just add this package to renv as well?  Then they just have to ask us to add new packages, which has the advantage that we can also 'consider carefully' ourselves (see text on next lines).

And to be honest, regarding the many difficulties I experienced to have these 'renv' packages installed in docker, and considering the fact that also less experienced R users may add protocols, I think it may be a good idea anyway to save novices from a headache add these packages ourselves.  As these are protocols, I expect mainly text with some illustrations or tables, so I suppose the list of packages is rather limited and will hardly grow after the first dozen of packages are added?  Or am I mistaken and could it be a large effort for us?

Anyway, to be added either as an issue, either as a discussion topic in the PR I submit to add the automation...

Reviewer: Sounds all very reasonable to me. I leave the further considerations to you and Hans.

        2.  alternatively, download the rendered version: see [these instructions](REVIEWING.md)

    5.  add text, media, ... to the Rmarkdown files
    
    6.  save your changes

    7.  stage, commit, push changes

8.  When finished, go to your draft pull request and press 'ready for review' 

   1. Wait for the continuous integration checks to finish and see if you the checks succeeded.

>    These checks will run `protocolhelper::check_frontmatter()` and `protocolhelper::check_structure()`.

Reviewer 2: ```suggestion
   These checks will run `protocolhelper::check_frontmatter()` and `protocolhelper::check_structure()`, and update the version number if needed.
```

>    In case there are problems with the YAML front matter of your `index.Rmd` file or problems with the structure of the protocol not confirming with the protocol templates, these problems will be listen and can be consulted.

Reviewer: > I'm not sure confirming is the right word here.

@ElsLommelen I've edited your suggestion

>        1.  Address the problems detected by `protocolhelper::check_frontmatter` or `protocolhelper::check_structure`. You can see the list of problems by clicking on the check online (at the bottom of the pull request webpage). Alternatively, you can run these functions locally and see the list of problems printed in your console.

Reviewer 2: ```suggestion
       1.  Address the problems detected by `protocolhelper::check_frontmatter` or `protocolhelper::check_structure`. You can see the list of problems by clicking on the check online (at the bottom of the pull request webpage). Alternatively, you can run these functions locally and see the list of problems printed in your console.
        Note that during this automatic run, a commit to update the version number may be automatically pushed to the remote.
        This should not pose a problem for local clones as this change can be automatically included and will not cause conflicts (i.e. `git pull origin main` with a local protocol branch checked out)
```

Reviewer: This should not pose a problem for local clones as this change can be automatically included and will not cause conflicts (i.e. \`git pull origin main\` with a local protocol branch checked out)

@ElsLommelen I don't understand what you mean with the last sentence. Anyway it may be too scary for contributors and perhaps unneeded, so it can be dropped?

Reviewer 2: I copied it more or less from a phrase of @hansvancalster in RELEASE.md, and it refers to the fact that the commit that is automatically generated in the GHA, should be pulled locally.  But you are right, it should be written less complicated.  What about something like `If this commit is added, press pull in the RStudio Git Pane to add it to your local clone.` to replace the last sentence?

As git often gives an error and asks to pull remote changes first before pushing additional commits, I would in any case write something, as this error may scare starting Github users as well.  (Or they may just click the message away without noticing their commit is not pushed to the origin.)

Reviewer: Your suggestion is OK indeed!

Now, it seems I was confused because the check-functions, either run in GHA or locally (line 122), cause a new commit to be 'pushed to the remote' (quoting your line 123 above). If that happened from the local repo, then pulling makes no sense. Maybe it doesn't happen that way when run locally, but that's not entirely clear here.

Reviewer 2: No, this new commit is pushed from the GHA to the remote by also running `update_version_number()` additionally to the check-functions.  So a local run of the check-functions will not cause a new commit (running `update_version_number()` locally will, but no need anymore to bother contributors with that and mention it in the guidelines).

Reviewer: Thanks for clarifying. Your suggested change (quoted below) is already sufficient to prevent further confusion, so fine to add that change!

> If this commit is added, press pull in the RStudio Git Pane to add it to your local clone.

Reviewer 2: edited above.
